### PR TITLE
Dm/da-588

### PR DESCRIPTION
**Pull request by DragosMoldovan**

### CODE REVIEW

File: marty_flutter/lib/screens/navigation/menu_app_bar.dart [hunk 1]
```diff
@@ -27,13 +28,13 @@ class MenuAppBar extends StatelessWidget {
...
-      duration: Duration(milliseconds: Values.menuNavigationAnimationDuration),
+      duration: Duration(milliseconds: AppValues.menuNavigationAnimationDuration),
       color: _openMenu ? Colors.transparent : AppColors.primaryColor,
-      height: Dimens.appBarHeight,
+      height: AppDimens.appBarHeight,
       child: Stack(
         children: <Widget>[
           Container(
-            margin: const EdgeInsets.only(left: Dimens.menuButtonMarginLeft),
+            margin: const EdgeInsets.only(left: AppDimens.menuButtonMarginLeft),
```
Comment (vladms) [inline marty_flutter/lib/screens/navigation/menu_app_bar.dart]: @DragosMoldovan Check the left padding of the button

File: marty_flutter/lib/screens/navigation/menu_app_bar.dart [hunk 2]
```diff
@@ -27,13 +28,13 @@ class MenuAppBar extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return AnimatedContainer(
-      duration: Duration(milliseconds: Values.menuNavigationAnimationDuration),
+      duration: Duration(milliseconds: AppValues.menuNavigationAnimationDuration),
       color: _openMenu ? Colors.transparent : AppColors.primaryColor,
-      height: Dimens.appBarHeight,
+      height: AppDimens.appBarHeight,
       child: Stack(
```
Comment (vladms) [inline marty_flutter/lib/screens/navigation/menu_app_bar.dart]: @DragosMoldovan Why do you use Stack here?

File: marty_flutter/lib/screens/navigation/top_navigation.dart [hunk 3]
```diff
@@ -10,42 +14,63 @@ class TopNavigation extends StatefulWidget {
...
+      );
 }
 
-class _TopNavigationState extends State<TopNavigation> {
+class _TopNavigationState extends SubscriptionState<TopNavigation> {
   final Function _menuPressed;
   final Function _navigateTo;
+  PublishSubject<MenuItemType> itemClickListener = PublishSubject();
+  TopNavigationViewModel _viewModel;
   bool _openMenu = false;
```
Comment (vladms) [inline marty_flutter/lib/screens/navigation/top_navigation.dart]: @DragosMoldovan Maybe rename to menuOpened ?

File: marty_flutter/lib/screens/navigation/menu_item_list.dart [hunk 4]
```diff
@@ -1,98 +1,107 @@
...
+    );
+  }
+}
+
+class _MenuItemListState extends SubscriptionState<MenuItemList> {
+  final Function _menuButtonPressed;
+  final Function _subMenuButtonPressed;
+  final TopNavigationViewModel _viewModel;
+  List<Widget> _listItems = [];
+  PublishSubject<MenuItemType> _itemClickListener;
```
Comment (vladms) [inline marty_flutter/lib/screens/navigation/menu_item_list.dart]: @DragosMoldovan If you do not change properties inside state then you can access them with `widget. _itemClickListener ` from the StatefullWidget

File: marty_flutter/pubspec.yaml [hunk 5]
```diff
@@ -20,6 +20,7 @@ dependencies:
   flutter:
     sdk: flutter
   configurable_expansion_tile: 1.0.0
```
Comment (vladms) [inline marty_flutter/pubspec.yaml]: @DragosMoldovan  Please implement your own expansion tile

File: marty_flutter/lib/screens/navigation/menu_item_list.dart [hunk 6]
```diff
@@ -1,98 +1,107 @@
...
   final Function _menuButtonPressed;
   final Function _subMenuButtonPressed;
+  final TopNavigationViewModel _viewModel;
+  final PublishSubject<MenuItemType> _itemClickListener;
 
   MenuItemList(
-    this._navigateTo,
     this._menuButtonPressed,
     this._subMenuButtonPressed,
+    this._viewModel,
```
Comment (vladms) [inline marty_flutter/lib/screens/navigation/menu_item_list.dart]: @DragosMoldovan  Why do you send the whole view model to MenuItemList?

File: marty_flutter/lib/screens/navigation/menu_item_list.dart [hunk 7]
```diff
@@ -1,98 +1,107 @@
...
+    super.initState();
+    _bindViewModel();
+  }
+
+  void _bindViewModel() {
+    disposeLater(_viewModel.output.menuItems.listen((listData) {
+      setState(() {
+        _listItems = listData.map((item) {
+          if (item.subList != null) {
+            return _ExpandableMenuItem(
```
Comment (vladms) [inline marty_flutter/lib/screens/navigation/menu_item_list.dart]: @DragosMoldovan Why do you need `callable`  when you have the type of each cell?

File: marty_flutter/lib/screens/navigation/menu_item_list.dart [hunk 8]
```diff
@@ -1,98 +1,107 @@
...
+              subItems: item.subList.map((subItem) {
+                return _MenuItemWidget(
+                  titleText: subItem.title,
+                  action: () => _itemClicked(item.type),
+                  isSubListItem: true,
+                );
+              }).toList(),
+              subMenuButtonPressed: _subMenuButtonPressed,
+            );
+          } else {
```
Comment (vladms) [inline marty_flutter/lib/screens/navigation/menu_item_list.dart]: @DragosMoldovan remove unnecessary `else`

File: marty_flutter/lib/screens/navigation/menu_item_list.dart [hunk 9]
```diff
@@ -102,49 +111,55 @@ class _LineDivider extends StatelessWidget {
...
     );
   }
 }
 
-class ExpandableMenuItem extends StatelessWidget {
-  final List<_MenuItem> _subItems;
-  final String _title;
-  final Function _subMenuButtonPressed;
+class _ExpandableMenuItem extends StatelessWidget {
+  final List<_MenuItemWidget> subItems;
```
Comment (vladms) [inline marty_flutter/lib/screens/navigation/menu_item_list.dart]: @DragosMoldovan Try to keep the same order for properties declaration and class constructor parameters

File: marty_flutter/lib/screens/navigation/menu_item_list.dart [hunk 10]
```diff
@@ -153,14 +168,89 @@ class ExpandableMenuItem extends StatelessWidget {
...
+        ],
+      ),
+    );
+  }
+}
+
+class _PhoneIcon extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Transform.rotate(
```
Comment (vladms) [inline marty_flutter/lib/screens/navigation/menu_item_list.dart]: @DragosMoldovan Why do you rotate the image?

File: marty_flutter/lib/screens/navigation/menu_item_list.dart [hunk 11]
```diff
@@ -153,14 +168,89 @@ class ExpandableMenuItem extends StatelessWidget {
...
+      angle: AppValues.phoneIconAngle,
+      child: IconButton(
+        icon: Icon(
+          Icons.phone,
+          color: AppColors.appWhite,
         ),
+        onPressed: null,
```
Comment (vladms) [inline marty_flutter/lib/screens/navigation/menu_item_list.dart]: @DragosMoldovan  No need too set onPressed

File: marty_flutter/lib/themes/dimens.dart [hunk 12]
```diff
@@ -9,7 +9,11 @@ class Dimens {
   static const double mainWidgetTopPadding_menuOpened = 475.0;
   static const double menuOpenTopPadding = 72.0;
 
-  static const double roundedCornersCircularRadius = 20.0;
+  static const double roundedCornersCircularRadius = 15.0;
```
Comment (vladms) [inline marty_flutter/lib/themes/dimens.dart]: @DragosMoldovan Rename this please. Maybe defaultCornerRadius?

File: marty_flutter/lib/themes/values.dart [hunk 13]
```diff
@@ -1,5 +1,6 @@
 
-class Values {
+class AppValues {
   static const int menuNavigationAnimationDuration = 200;
+  static const double phoneIconAngle = 270 * 3.14 / 180;
```
Comment (vladms) [inline marty_flutter/lib/themes/values.dart]: @DragosMoldovan  ??

File: marty_flutter/lib/screens/navigation/top_navigation.dart [hunk 14]
```diff
@@ -10,42 +14,63 @@ class TopNavigation extends StatefulWidget {
...
+  }
+
+  void _menuButtonPressed() {
     setState(
       () {
         numberOfExpandedItems = 0;
         _openMenu = !_openMenu;
         if (_openMenu) {
-          _menuPressed(Dimens.mainWidgetTopPadding_menuOpened);
+          _menuPressed(AppDimens.mainWidgetTopPadding_menuOpened);
```
[the remark this answers is not in the document]
Comment (DragosMoldovan) [reply marty_flutter/lib/screens/navigation/top_navigation.dart]: @vladms only in this file.

**Pull request merged**